### PR TITLE
[plugin.audio.ampache] 1.2.0

### PR DESCRIPTION
**Pull request by lusum**
### Description

New plugin, official ampache server/nexcloud music plugin for kodi

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.audio.ampache/addon.xml
Comment (enen92) [inline plugin.audio.ampache/addon.xml:15]: missing license tag `<license>GPL-2.0-or-later</license>`

File: plugin.audio.ampache/addon.xml
Comment (enen92) [inline plugin.audio.ampache/addon.xml:18]: since you'll have to touch this file please reindent it properly

File: plugin.audio.ampache/addon.xml
Comment (enen92) [inline plugin.audio.ampache/addon.xml:9]: start argument was removed, please remove

File: plugin.audio.ampache/addon.xml
Comment (enen92) [inline plugin.audio.ampache/addon.xml:4]: missing xbmc.python dependency

File: plugin.audio.ampache/default.py
Comment (enen92) [inline plugin.audio.ampache/default.py:240]: returning here doesn't seem to be necessary

File: plugin.audio.ampache/default.py
Comment (enen92) [inline plugin.audio.ampache/default.py:327]: please use setResolvedUrl, plugins are not allowed to call Player.play directly

File: plugin.audio.ampache/default.py
Comment (enen92) [inline plugin.audio.ampache/default.py:1069]: usually it is a good practice to keep the entrypoint file (the one mentioned in addon.xml) as smaller as possible as it is the only file that is not compiled to bytecode:

```python
from myplugin import Main
Main()
```

File: plugin.audio.ampache/resources/lib/ampache_connect.py
Comment (enen92) [inline plugin.audio.ampache/resources/lib/ampache_connect.py:142]: please replace all entries of `xbmc.executebuiltin(amp_notif)` by the python api equivalent: `xbmcgui.Dialog().notification()`

File: plugin.audio.ampache/resources/lib/art.py
Comment (enen92) [inline plugin.audio.ampache/resources/lib/art.py:66]: you never close the file handler. Use python's with statement:

```
with open(...) as f:
 f.write(contents)
```

File: plugin.audio.ampache/resources/lib/service.py
Comment (enen92) [inline plugin.audio.ampache/resources/lib/service.py:48]: you're not filtering for files started from your plugin (`plugin://plugin.audio.ampache...`) so this will be executed for any file kodi plays...

**Review by enen92 — CHANGES_REQUESTED** [review]

File: plugin.audio.ampache/addon.xml
Comment (enen92) [reply plugin.audio.ampache/addon.xml:9]: disregard, this only applies to matrix

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:15]: Ok, i will add this

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:18]: Ok, i will indent

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:4]: Ok, i will add

File: plugin.audio.ampache/default.py
Comment (lusum) [reply plugin.audio.ampache/default.py:240]: I will remove

File: plugin.audio.ampache/default.py
Comment (lusum) [reply plugin.audio.ampache/default.py:327]: AmpachePlayer is a class built in the plugin that use setResolvedUrl, as you say, i have added it to add some checks if it is necessary

EDIT: Ok, it was useless, so i have get rid of it

File: plugin.audio.ampache/default.py
Comment (lusum) [reply plugin.audio.ampache/default.py:1069]: This is a big change, i have to close this push, test it and release a new version

File: plugin.audio.ampache/resources/lib/ampache_connect.py
Comment (lusum) [reply plugin.audio.ampache/resources/lib/ampache_connect.py:142]: Ok, no problem

File: plugin.audio.ampache/resources/lib/art.py
Comment (lusum) [reply plugin.audio.ampache/resources/lib/art.py:66]: Ok

File: plugin.audio.ampache/resources/lib/service.py
Comment (lusum) [reply plugin.audio.ampache/resources/lib/service.py:48]: I have no idea how to filter for files, i have found very little documentation on this topic, some suggestion?

**Comment by lusum** [comment]
Ok, checked the errors, i will correct. When the changes to do will be clear and tested, i will close this push and i will submit another one corrected

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:18]: Btw, i have checked other plugins and it is indented in the same way, maybe you can give me an example of a plugin with a correct indentation?

Comment (lusum) [reply plugin.audio.ampache/addon.xml:18]: Ok, found the problem

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:15]: done

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:18]: done

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:4]: done

File: plugin.audio.ampache/default.py
Comment (lusum) [reply plugin.audio.ampache/default.py:240]: done

File: plugin.audio.ampache/default.py
Comment (lusum) [reply plugin.audio.ampache/default.py:1069]: done

File: plugin.audio.ampache/resources/lib/art.py
Comment (lusum) [reply plugin.audio.ampache/resources/lib/art.py:66]: done

File: plugin.audio.ampache/resources/lib/ampache_connect.py
Comment (lusum) [reply plugin.audio.ampache/resources/lib/ampache_connect.py:142]: done

File: plugin.audio.ampache/addon.xml
Comment (lusum) [reply plugin.audio.ampache/addon.xml:4]: i have added xbmc.python dependency 2.26.0, but now it doesn't work on kodi 19 ( works in kodi 18 btw )

Comment (pkscout) [reply plugin.audio.ampache/addon.xml:4]: You have to have a different addon.xml for Leia and Matrix because of the change in Python versions (Python 2.x in Leia, Python 3.x in Matrix onward). So for the version you've submitted to the Leia branch, you need the 2.26.0 Python dependency.  You will have to submit separately to the Matrix repo, and that addon.xml has to have the 3.0.0  Python dependency.  It also has to have a difference version number (greater than what is in Leia). We suggest adding `+matrix.1` to the version for Matrix.

Comment (lusum) [reply plugin.audio.ampache/addon.xml:4]: In my repository i skip xbmc.python to mantain full compatibility, but if i need to add different addon.xml files i don't know how to do. I have to create different branches and merge them every time i made a commit or i simply add an addon.xml+matrix.1 file as you seems to suggest? Correct me if i'm wrong

EDIT:ok, maybe i have understood, i can have a neutral addon.xml in my repository, but i will add xmbc.python requirement when i will submit the pull

Comment (pkscout) [reply plugin.audio.ampache/addon.xml:4]: We require the Python dependency for the official repo.  For addons that are 100% the same code between Python2 and Python3, what I do is keep one branch on Github. When I'm ready to submit to the official repo, I submit to the Matrix branch first (with the addon.xml updated as needed), and then submit to the Leia branch after the Matrix one is approved.  For cases where there are code differences, I maintain two branches and manage them separately. We understand this is a bit of a pain, but this is the only time you'll have to do this between versions. Well, at least until we have to go to Python4.  '-)

Comment (lusum) [reply plugin.audio.ampache/addon.xml:4]: Ok, i will create two different branches and i will merge the diffs when i will submit on kodi repository, yes, a little bit a pain, but i think it is manegeable. Now i have only a couple of changes to do, i will wait some tips about my AmpachePlayer class ( not so necessary , so i can get rid of it if it become a problem ) and the check that the files played are from my plugin ( honestly i don't know how to do, the messages i receive from kodi don't include the  plugin source )

EDIT: created the different branches

File: plugin.audio.ampache/default.py
Comment (lusum) [reply plugin.audio.ampache/default.py:327]: Ok, not perfect, but done

**Comment by lusum** [comment]
i have modified the plugin following all the changes requested, now i will close this pull and i will submit another one.

File: plugin.audio.ampache/resources/lib/service.py
Comment (lusum) [reply plugin.audio.ampache/resources/lib/service.py:48]: Not perfect, but corrected

**Pull request closed without merging**
